From fdde2d099aa0ce1700380e4cded1ef47d0689cb3 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Wed, 21 Feb 2024 16:30:47 +0100
Subject: [PATCH 2/2] Link to absl_hash

---
 cmake/protobuf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index 805aa784..e03ac8f6 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -186,7 +186,7 @@ MACRO(MYSQL_CHECK_PROTOBUF)
   IF(WIN32)
     SET(protobuf_dependencies abseil_dll)
   ELSE()
-    SET(protobuf_dependencies absl_spinlock_wait)
+    SET(protobuf_dependencies absl_spinlock_wait absl_hash)
   ENDIF()
   SET_TARGET_PROPERTIES(ext::libprotobuf PROPERTIES
     INTERFACE_LINK_LIBRARIES "${protobuf_dependencies}"
