From d4461481961311a0f2473e8a771d93888fc35fc5 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 22 Jun 2020 09:53:55 -0400
Subject: [PATCH 03/23] Allow flags other than macros in CPPFLAGS

---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4bc1d51c..0d381d10 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -690,7 +690,9 @@ ENDFOREACH()
 # Following autotools tradition, add preprocessor definitions
 # specified in environment variable CPPFLAGS
 IF(DEFINED ENV{CPPFLAGS})
-  ADD_DEFINITIONS($ENV{CPPFLAGS})
+  SET(CPPFLAGS $ENV{CPPFLAGS})
+  SEPARATE_ARGUMENTS(CPPFLAGS)
+  ADD_COMPILE_OPTIONS(${CPPFLAGS})
 ENDIF()
 
 # To add c/c++ code specific for MySQL AI release using #ifdef..#endif
