From 51318a6b883459476de6911866a899bf5f2621e6 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 19 Oct 2020 23:48:16 -0400
Subject: [PATCH 04/25] Add include header for std::string

---
 sql/create_def.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sql/create_def.cc b/sql/create_def.cc
index 5a7e64a4..d291b2a1 100644
--- a/sql/create_def.cc
+++ b/sql/create_def.cc
@@ -68,6 +68,7 @@
 #include <thread>
 #include <unordered_set>
 #include <vector>
+#include <string>
 
 /** Prints an error message supplied and attaches GetLastError with formatted
  * message. */
