From ac543f5ff316e57644375552d593769b2fbc3072 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 23 Jul 2020 12:56:13 -0400
Subject: [PATCH 05/21] Remove forceful injection of debug info

---
 cmake/os/Windows.cmake | 17 +++++++----------
 1 file changed, 7 insertions(+), 10 deletions(-)

diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index f13bbdf5fff..e8907687900 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -157,13 +157,13 @@ IF(MSVC)
   #     Allowing inline reduces test time using the debug server by
   #     30% or so. If you do want to keep inlining off, set the
   #     cmake flag WIN_DEBUG_NO_INLINE.
-  FOREACH(lang C CXX)
-    IF(FPROFILE_GENERATE OR FPROFILE_USE)
-      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Zi")
-    ELSE()
-      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Z7")
-    ENDIF()
-  ENDFOREACH()
+  #FOREACH(lang C CXX)
+  #  IF(FPROFILE_GENERATE OR FPROFILE_USE)
+  #    SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Zi")
+  #  ELSE()
+  #    SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Z7")
+  #  ENDIF()
+  #ENDFOREACH()
 
   FOREACH(flag
       CMAKE_C_FLAGS_MINSIZEREL
@@ -177,9 +177,6 @@ IF(MSVC)
     ENDIF()
     IF(FPROFILE_GENERATE OR FPROFILE_USE)
       STRING(REPLACE "/ZI"  "/Zi" "${flag}" "${${flag}}")
-    ELSE()
-      STRING(REPLACE "/Zi"  "/Z7" "${flag}" "${${flag}}")
-      STRING(REPLACE "/ZI"  "/Z7" "${flag}" "${${flag}}")
     ENDIF()
     IF (NOT WIN_DEBUG_NO_INLINE)
       STRING(REPLACE "/Ob0"  "/Ob1" "${flag}" "${${flag}}")
-- 
2.45.2

