From 42f63389fa3bf28df2c6a158f5a474a4bcec9807 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Thu, 23 Jul 2020 12:56:13 -0400
Subject: [PATCH 06/24] Remove forceful injection of debug info

---
 cmake/os/Windows.cmake | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/cmake/os/Windows.cmake b/cmake/os/Windows.cmake
index 8ae3d150..a1726cbc 100644
--- a/cmake/os/Windows.cmake
+++ b/cmake/os/Windows.cmake
@@ -151,13 +151,13 @@ IF(MSVC)
   #     Allowing inline reduces test time using the debug server by
   #     30% or so. If you do want to keep inlining off, set the
   #     cmake flag WIN_DEBUG_NO_INLINE.
-  FOREACH(lang C CXX)
-    IF(FPROFILE_GENERATE OR FPROFILE_USE)
-      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Zi")
-    ELSE()
-      SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Z7")
-    ENDIF()
-  ENDFOREACH()
+  #FOREACH(lang C CXX)
+  #  IF(FPROFILE_GENERATE OR FPROFILE_USE)
+  #    SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Zi")
+  #  ELSE()
+  #    SET(CMAKE_${lang}_FLAGS_RELEASE "${CMAKE_${lang}_FLAGS_RELEASE} /Z7")
+  #  ENDIF()
+  #ENDFOREACH()
 
   FOREACH(flag
       CMAKE_C_FLAGS_MINSIZEREL
@@ -171,10 +171,10 @@ IF(MSVC)
     ENDIF()
     IF(FPROFILE_GENERATE OR FPROFILE_USE)
       STRING(REPLACE "/ZI"  "/Zi" "${flag}" "${${flag}}")
-    ELSE()
-      STRING(REPLACE "/Zi"  "/Z7" "${flag}" "${${flag}}")
-      STRING(REPLACE "/ZI"  "/Z7" "${flag}" "${${flag}}")
-    ENDIF()
+    #ELSE()
+    #  STRING(REPLACE "/Zi"  "/Z7" "${flag}" "${${flag}}")
+    #  STRING(REPLACE "/ZI"  "/Z7" "${flag}" "${${flag}}")
+    #ENDIF()
     IF (NOT WIN_DEBUG_NO_INLINE)
       STRING(REPLACE "/Ob0"  "/Ob1" "${flag}" "${${flag}}")
     ENDIF()
