From c34db22c255b29d69e1ffcb7ff57f8bae9a3fee5 Mon Sep 17 00:00:00 2001
From: Nehal J Wani <nehaljw.kkd1@gmail.com>
Date: Mon, 19 Jun 2023 20:25:57 +0100
Subject: [PATCH 17/23] Make libfido checks conditional

---
 CMakeLists.txt | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2b0c86e..1f4dd91c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2082,8 +2082,10 @@ ENDIF()
 MYSQL_CHECK_RAPIDJSON()
 
 # Look for fido2 library
-MYSQL_CHECK_FIDO()
-MYSQL_CHECK_FIDO_DLLS()
+IF(WITH_AUTHENTICATION_CLIENT_PLUGINS)
+  MYSQL_CHECK_FIDO()
+  MYSQL_CHECK_FIDO_DLLS()
+ENDIF()
 
 IF(APPLE)
   GET_FILENAME_COMPONENT(HOMEBREW_BASE ${HOMEBREW_HOME} DIRECTORY)
