From 0177c62acdd0994b4a19a8601b294802f3204b5a Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Tue, 30 Jan 2024 17:21:06 +0100
Subject: [PATCH 23/23] Use fully-fledged FindProtobuf

---
 cmake/protobuf.cmake | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index d470de5f..d50b124f 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -119,11 +119,8 @@ MACRO(MYSQL_CHECK_PROTOBUF)
   IF(WITH_PROTOBUF STREQUAL "bundled")
     MYSQL_USE_BUNDLED_PROTOBUF()
   ELSE()
-      FIND_PACKAGE(Protobuf CONFIG)
+      FIND_PACKAGE(Protobuf)
       IF(Protobuf_FOUND)
-          SET(PROTOBUF_LIBRARY protobuf::libprotobuf)
-          SET(PROTOBUF_PROTOC_LIBRARY protobuf::libprotoc)
-          SET(PROTOBUF_LITE_LIBRARY protobuf::libprotobuf-lite)
           IF(NOT PROTOBUF_PROTOC_EXECUTABLE)
               SET(PROTOBUF_PROTOC_EXECUTABLE protobuf::protoc)
           ENDIF()
-- 
2.39.3 (Apple Git-145)

