From cde1f6d968e3ad9864c0810389040d4cbf885f04 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Fri, 2 Feb 2024 16:56:09 +0100
Subject: [PATCH 24/26] Link to absl_spinlock_wait

---
 cmake/protobuf.cmake | 32 +++++++++++++-------------------
 1 file changed, 13 insertions(+), 19 deletions(-)

diff --git a/cmake/protobuf.cmake b/cmake/protobuf.cmake
index 84039b5a..e03ac8f6 100644
--- a/cmake/protobuf.cmake
+++ b/cmake/protobuf.cmake
@@ -183,26 +183,20 @@ MACRO(MYSQL_CHECK_PROTOBUF)
 
   FIND_PROTOBUF_VERSION()
 
-  # Version 22 and up depend on ~65 abseil .dylibs.
-  IF(APPLE AND WITH_PROTOBUF STREQUAL "system" AND
-      PB_MINOR_VERSION VERSION_GREATER 21)
-    # list(FILTER <list> {INCLUDE | EXCLUDE} REGEX <regex>)
-    FIND_OBJECT_DEPENDENCIES("${PROTOBUF_LIBRARY}" protobuf_dependencies)
-    LIST(FILTER protobuf_dependencies INCLUDE REGEX "${HOMEBREW_HOME}.*")
-    SET_TARGET_PROPERTIES(ext::libprotobuf PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${protobuf_dependencies}"
-      )
-    FIND_OBJECT_DEPENDENCIES("${PROTOBUF_LITE_LIBRARY}" lite_dependencies)
-    LIST(FILTER lite_dependencies  INCLUDE REGEX "${HOMEBREW_HOME}.*")
-    SET_TARGET_PROPERTIES(ext::libprotobuf-lite PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${lite_dependencies}"
-      )
-    FIND_OBJECT_DEPENDENCIES("${Protobuf_PROTOC_LIBRARY}" protoc_dependencies)
-    LIST(FILTER protoc_dependencies INCLUDE REGEX "${HOMEBREW_HOME}.*")
-    SET_TARGET_PROPERTIES(ext::libprotoc PROPERTIES
-      INTERFACE_LINK_LIBRARIES "${protoc_dependencies}"
-      )
+  IF(WIN32)
+    SET(protobuf_dependencies abseil_dll)
+  ELSE()
+    SET(protobuf_dependencies absl_spinlock_wait absl_hash)
   ENDIF()
+  SET_TARGET_PROPERTIES(ext::libprotobuf PROPERTIES
+    INTERFACE_LINK_LIBRARIES "${protobuf_dependencies}"
+    )
+  SET_TARGET_PROPERTIES(ext::libprotobuf-lite PROPERTIES
+    INTERFACE_LINK_LIBRARIES "${protobuf_dependencies}"
+    )
+  SET_TARGET_PROPERTIES(ext::libprotoc PROPERTIES
+    INTERFACE_LINK_LIBRARIES "${protobuf_dependencies}"
+    )
 
   IF("${PROTOBUF_VERSION}" VERSION_LESS "${MIN_PROTOBUF_VERSION_REQUIRED}")
     COULD_NOT_FIND_PROTOBUF()
